File: LWHeroOC/Classes/BasePreprocessor.m
//
//  BasePreprocessor.m
//  Hero-ObjectiveC
//
//  Created by luca.li on 2017/1/22.
//  Copyright © 2017年 luca. All rights reserved.
//

#import "BasePreprocessor.h"
#import "Hero.h"

@implementation BasePreprocessor

- (HeroContext *)context {
    return [Hero shared].context;
}

- (void)processFromViews:(NSArray *)fromviews toViews:(NSArray *)toviews {
    return;
}
@end

File: LWHero_swift/LWHero/Classes/BasePreprocessor.swift
//
//  BasePreprocessor.swift
//  LWHero
//
//  Created by Swift Port
//  Copyright © 2025. All rights reserved.
//

import Foundation
import UIKit

open class BasePreprocessor: NSObject, HeroPreprocessor {
    // MARK: - Properties

    public private(set) var context: HeroContext?

    // MARK: - Initialization

    public override init() {
        super.init()
    }

    // MARK: - HeroPreprocessor

    open func process(fromViews: [UIView], toViews: [UIView]) {
        // Override in subclass
    }
}
